ui/pages/settings/contact-list-tab/contact-list-tab.component.js

```diff
@@ -1,16 +1,57 @@
 import React, { Component } from 'react';
+import { useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
 import ContactList from '../../../components/app/contact-list';
-import {
-  CONTACT_ADD_ROUTE,
-  CONTACT_VIEW_ROUTE,
-} from '../../../helpers/constants/routes';
-import Button from '../../../components/ui/button';
+import { showModal } from '../../../store/actions';
+import { CONTACT_VIEW_ROUTE } from '../../../helpers/constants/routes';
 import EditContact from './edit-contact';
 import AddContact from './add-contact';
 import ViewContact from './view-contact';
 
+AddContactButton.propTypes = {
+  label: PropTypes.string,
+};
+
+RenderAddContactLink.propTypes = {
+  label: PropTypes.string,
+};
+
+function RenderAddContactLink(props) {
+  const { label } = props;
+  const dispatch = useDispatch();
+
+  return (
+    <div>
+      <button
+        className="address-book__link"
+        onClick={(e) => {
+          e.preventDefault();
+          dispatch(showModal({ name: 'ADD_NEW_CONTACT' }));
```

Comment:
When the add new contact modal is a popover, it can be rendered inline and only displayed when a state variable changes. So instead of having to dispatch a showModal, you can just set a state variable that gets passed to the Popover. This will help keep more of the logic around adding a contact in roughly the same place. It's also fine to separate them into different components.